File: C/Stack_10828.cpp
#include <stdio.h>
#include <string.h>
#include <stack>

int main() {
    std::stack<int> s;
    int N;
    // freopen("input.txt", "r", stdin);
    scanf("%d", &N);

    char cmd[100]; int value;
    for(int i=0; i<N; i++) {
        scanf("%s", cmd);
        if(strcmp(cmd, "push") == 0) {
            scanf("%d", &value);
            s.push(value);
        } else if(strcmp(cmd, "top") == 0) {
            if(s.size() == 0)   printf("-1\n");
            else                printf("%d\n", s.top());
        } else if(strcmp(cmd, "pop") == 0) {
            if(s.size() == 0)   printf("-1\n");
            else {
                printf("%d\n", s.top());
                s.pop();
            }
        } else if(strcmp(cmd, "empty") == 0) {
            if(s.size() == 0)   printf("1\n");
            else                printf("0\n");
        } else if(strcmp(cmd, "size") == 0) {
            printf("%d\n", (int)s.size());
        }
    }
    
    return 0;
}

File: Python/Stack_10828.py
def main():
    # 입력 처리
    import sys
    N = int(sys.stdin.readline())
    data = [list(map(str, input().split())) for _ in range(N)]
    stack = [];

    for i in range(0, N):
        if(data[i][0] == "push"):
            stack.append(data[i][1])
        if(data[i][0] == "pop"):
            if(len(stack) == 0): 
                print(-1)
            else:
                print(stack[-1])
                stack.pop()
        if(data[i][0] == "top"):
            if(len(stack) == 0): 
                print(-1)
            else:
                print(stack[-1])
        if(data[i][0] == "size"):
            print(len(stack))
        if(data[i][0] == "empty"):
            if(len(stack) == 0): 
                print(1)
            else:
                print(0)

if __name__ == "__main__":
    main()